extensions/scarb-cairo-run/src/main.rs

```diff
@@ -41,7 +45,13 @@ fn main() -> Result<()> {
 
     let package = args.packages_filter.match_one(&metadata)?;
 
-    ScarbCommand::new().arg("build").run()?;
+    if !args.no_build {
+        let filter = PackagesFilter::generate_for::<Metadata>(vec![package.clone()].iter());
+        ScarbCommand::new()
+            .arg("build")
+            .env("SCARB_PACKAGES_FILTER", filter.to_env())
```

Comment:
Why is this env now set?

Reply:
Just as we do in cairo test here: https://github.com/software-mansion/scarb/blob/31b92d4cd940539fa47b800571d2e9c8b329d97d/extensions/scarb-cairo-test/src/main.rs#L38-L43
It's used to override the PackagesFilter in the subsequent call to scarb binary. 
https://github.com/software-mansion/scarb/blob/31b92d4cd940539fa47b800571d2e9c8b329d97d/utils/scarb-ui/src/args/packages_filter.rs#L32

The PackagesFilter controls which package is build, via both `--package/--workspace` arguments, current working directory and some logic for workspace package selection. 
User can run the cairo-run extension from some package in the workspace, passing another package's name to `--package` argument. In this case, we want to build the package that has been selected by name, but simply calling `ScarbCommand` would result in building package from cwd instead!

As the subcommands are responsible for package selection on their own (in contrary to scripts, which are run in a select package / group of packages) (which cairo-run does but consuming the --package/--workspace flags, which are ignored by Scarb itself), the responsibility for this also needs to lay here.